skfda/representation/irregular.py

```diff
@@ -379,32 +465,22 @@ def from_fdatagrid(
 
     def _sort_by_arguments(self) -> Tuple[ArrayLike, ArrayLike]:
         """Sort the arguments lexicographically functionwise.
-
+    
         Additionally, sort the values accordingly.
-
+    
         Returns:
             Tuple[ArrayLike, Arraylike]: sorted pair (arguments, values)
         """
-        slice_args = np.split(self.points, self.start_indices[1:])
-        slice_values = np.split(self.values, self.start_indices[1:])
-
-        # Sort lexicographically, first to last dimension
-        sorting_masks = [
-            np.lexsort(np.flip(f_args, axis=1).T)
-            for f_args in slice_args
-        ]
-
-        sorted_args = [
-            slice_args[i][mask]
-            for i, mask in enumerate(sorting_masks)
-        ]
-
-        sorted_values = [
-            slice_values[i][mask]
-            for i, mask in enumerate(sorting_masks)
-        ]
+        points_split = np.split(self.points, self.start_indices)[1:]
+        shifts = itertools.accumulate(map(len, [[]] + points_split[:-1]))
+        sorter = np.concatenate(
+            [
+                np.lexsort(np.rot90(points)) + shift
```

Comment:
I honestly do not get what `np.rot90` is doing here. Can't we use the `axis` parameter of `np.lexsort` instead?

Reply:
I'm not sure, I think this is due to `lexsort` being written in a weird way, reading lines or columns from the bottom from what I remember.

In the original implementation, it was written `np.lexsort(np.flip(f_args, axis=1).T)`, which I think was pretty opaque, I simply replaced this with `np.lexsort(np.rot90(points))` so that at least we know what's happening to `points` (90 degres rotation of the matrix) at first glance.

Reply:
![image](https://github.com/GAA-UAM/scikit-fda/assets/121242234/51f67b96-09ab-4fdd-bed4-7d68ac70d668)

Reply:
But why not just transposing then?

Reply:
Because it won't sort as expected:
```pycon
>>> t = np.random.randint(0,5,(10,3))
>>> t
array([[1, 0, 3],
       [0, 2, 3],
       [0, 0, 0],
       [3, 4, 3],
       [3, 0, 2],
       [3, 3, 3],
       [4, 4, 0],
       [4, 1, 2],
       [3, 0, 1],
       [0, 1, 4]])
>>> t[np.lexsort(t.T)]
array([[0, 0, 0],
       [4, 4, 0],
       [3, 0, 1],
       [3, 0, 2],
       [4, 1, 2],
       [1, 0, 3],
       [0, 2, 3],
       [3, 3, 3],
       [3, 4, 3],
       [0, 1, 4]])
>>> t[np.lexsort(np.rot90(t))]
array([[0, 0, 0],
       [0, 1, 4],
       [0, 2, 3],
       [1, 0, 3],
       [3, 0, 1],
       [3, 0, 2],
       [3, 3, 3],
       [3, 4, 3],
       [4, 1, 2],
       [4, 4, 0]])
```

Reply:
Ok, I get it. `lexsort` uses the keys in DECREASING order, from last to first (maybe because that is faster in C-order?), and also expect them as the first axis.

Thus, for me the best way around is doing:
`t[np.lexsort(t.T[::-1])]`

and putting an explanatory comment. I do not think `np.rot90` is very clear, as it is semantically detached from what we want here (although it ends being the same). The need for transposition and flip should be clear from the comment. The syntax for the flip over the first axis should also be clear (is the same as for lists). We could also use `t[np.lexsort(np.flip(t.T, axis=0))]` if you think that is more legible, but I have my doubts.

Reply:
I think, as you point out, that any formulation requires a comment. Without any comment, `t[np.lexsort(t.T[::-1])]` is not any clearer to me.

As such, I personally would prefer something like
```
sorter = np.lexsort(np.rot90(np.c_[ind, points]))  # rot90 for lexsort
```
over
```
sorter = np.lexsort(np.c_[ind, points].T[::-1])  # <comment>
```

I don't mind if you prefer the latter, but in this case, can you provide a comment you'd like? Ideally, the whole line should be <72 characters :)

Reply:
Ok, thanks for providing your comment, I will then implement your proposition when I get around my pc :)

I did not know about the standardization effort, it's interesting! But I wonder then if wrapping a ufunc was a good way to go forward.

Reply:
> I did not know about the standardization effort, it's interesting! But I wonder then if wrapping a ufunc was a good way to go forward.

I think that in the future we could provide a "fast" NumPy path in that case and a slower implementation for everything else. At least I think in SciPy they are doing that in some places.

Reply:
> I would add the following comment above the code, instead of on the side.

Done :)